commons/all/src/main/java/org/infinispan/commons/dataconversion/JavaObjectCodec.java

```diff
@@ -0,0 +1,113 @@
+package org.infinispan.commons.dataconversion;
+
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_OBJECT;
+
+/**
+ * Encodes/decodes Java primitives, byte arrays and String as text.
+ *
+ * <br/> It only applies to {@link MediaType#APPLICATION_OBJECT} content with a "type" parameter.
+ *
+ * The following values are supported for the "type" param:
+ * <br/>
+ * <ul>
+ *    <li>java.lang.String</li>
+ *    <li>java.lang.String</li>
+ *    <li>java.lang.Boolean</li>
+ *    <li>java.lang.Short</li>
+ *    <li>java.lang.Byte</li>
+ *    <li>java.lang.Integer</li>
+ *    <li>java.lang.Long</li>
+ *    <li>java.lang.Float</li>
+ *    <li>java.lang.Double</li>
+ *    <li>ByteArray</li>
+ * </ul>
+ *
+ * When "ByteArray" is used, it implies the content is encoded as a <a href="https://www.ietf.org/rfc/rfc4648.txt">RFC 4648</a>
+ * hexadecimal prefixed by "0x". For all other supported types, it will use {@link Object#toString()}.
+ *
+ * @since 13.0
+ */
+class JavaObjectCodec implements MediaTypeCodec {
+
+   @Override
+   public Object decodeContent(Object content, MediaType contentType) {
+      String type = contentType.getClassType();
+      if (!contentType.match(APPLICATION_OBJECT) || type == null) {
+         return content;
+      }
+      String strContent;
+      if (content instanceof byte[]) {
+         strContent = new String((byte[]) content, contentType.getCharset());
+      } else if (content instanceof String) {
+         strContent = content.toString();
+      } else {
+         return content;
+      }
+      switch (type) {
+         case "ByteArray":
```

Comment:
could we have an enum or something and use it in the 'type' from the contentType class type ?